plugins/module_utils/foreman_helper.py

```diff
@@ -282,15 +282,24 @@ def __init__(self, **kwargs):
             lifecycle_environment=dict(type='entity', scope=['organization']),
             kickstart_repository=dict(type='entity', scope=['organization'], resource_type='repositories'),
             content_view=dict(type='entity', scope=['organization']),
+            activation_keys=dict(),
         )
         foreman_spec.update(kwargs.pop('foreman_spec', {}))
         required_plugins = kwargs.pop('required_plugins', []) + [
-            ('katello', ['content_source', 'lifecycle_environment', 'kickstart_repository', 'content_view']),
+            ('katello', ['activation_keys', 'content_source', 'lifecycle_environment', 'kickstart_repository', 'content_view']),
             ('openscap', ['openscap_proxy']),
         ]
         mutually_exclusive = kwargs.pop('mutually_exclusive', []) + [['medium', 'kickstart_repository']]
         super(HostMixin, self).__init__(foreman_spec=foreman_spec, required_plugins=required_plugins, mutually_exclusive=mutually_exclusive, **kwargs)
 
+    def run(self, **kwargs):
+        if 'activation_keys' in self.foreman_params:
+            if 'parameters' not in self.foreman_params:
+                self.foreman_params['parameters'] = []
+            self.foreman_params['parameters'].append({'name': 'kt_activation_keys', 'value': self.foreman_params.pop('activation_keys')})
```

Comment:
This looks like the `kt_activation_keys` could end up twice in this list. :exploding_head:

Reply:
Same as when the user provides the same param twice: 🤯

Reply:
So, not a new problem here. :heavy_check_mark:

Reply:
given the code, and #1106 - I don't think the check from #1106 will catch that :(

Reply:
Because the check will run as the last part of the `super().__init__()`.
Too bad.
I put the check in the `__init__` to check before we perform the ping. But maybe we need to move it into `run`.

Reply:
followup in https://github.com/theforeman/foreman-ansible-modules/pull/1114